.github/workflows/phpunit-mariadb.yml

```
@@ -0,0 +1,146 @@
+        uses: shivammathur/setup-php@4bd44f22a98a19e0950cbad5f31095157cc9621b # v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          # https://docs.nextcloud.com/server/stable/admin_manual/installation/source_installation.html#prerequisites-for-manual-installation
+          extensions: bz2, ctype, curl, dom, fileinfo, gd, iconv, intl, json, libxml, mbstring, openssl, pcntl, posix, session, simplexml, xmlreader, xmlwriter, zip, zlib, mysql, pdo_mysql
+          coverage: none
+          ini-file: development
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up dependencies
+        run: composer i
+
+      - name: Set up Nextcloud
+        env:
+          DB_PORT: 4444
+        run: |
+          mkdir data
+          echo -e "<?php\n\$CONFIG = ['memcache.locking' => '\\OC\\Memcache\\Redis', 'memcache.distributed' => '\\OC\\Memcache\\Redis','memcache.local' =>'\\OC\Memcache\\Redis','redis' => ['host' => 'localhost', 'port' => 6379]];" > config/config.php
+          ./occ maintenance:install --verbose --database=mysql --database-name=nextcloud --database-host=127.0.0.1 --database-port=$DB_PORT --database-user=root --database-pass=rootpassword --admin-user admin --admin-pass admin
+
+      - name: Wait for MariaDB
+        id: check_mariadb_up
+        run: |
+          apps/files_external/tests/env/wait-for-connection localhost 4444 30
+
+      - name: Enable ONLY_FULL_GROUP_BY MySQL option
+        if: steps.check_mariadb_up.outcome == 'success'
+        run: |
+          echo "SET GLOBAL sql_mode=(SELECT CONCAT(@@sql_mode,',ONLY_FULL_GROUP_BY'));" | mysql -h 127.0.0.1 -P 4444 -u root -prootpassword
+          echo "SELECT @@sql_mode;" | mysql -h 127.0.0.1 -P 4444 -u root -prootpassword
+
+      - name: Run Nextcloud
+        if: steps.check_mariadb_up.outcome == 'success'
+        run: php -S localhost:8080 &
+
+      - name: PHPUnit
+        if: steps.check_mariadb_up.outcome == 'success'
```

So if mariadb did not start, we don't run tests :D
By skipping this the end result will be "pass"?

---

No it will fail because the last step (check_mariadb_up) failed: failed -> skipped -> skipped => failed.

---

I still find this unnecessary complex and would prefer to fail all the steps instead of skipping them 🙈